src/common/utility.cpp

```diff
@@ -0,0 +1,66 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// utility.cpp
+//
+// Identification: src/include/common/utility.h
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include <unistd.h>
+#include <cerrno>
+#include <string>
+#include <vector>
+
+#include "common/utility.h"
+#include "loggers/main_logger.h"
+#if __APPLE__
+extern "C" {
+#include <sys/cdefs.h>
+int close$NOCANCEL(int);
+};
+#endif
+
+namespace terrier {
+
+int peloton_close(int fd) {
```

Comment:
We should probably retrofit this file a little. For this particular function:

- If the return code is not supposed to be checked, maybe there's no reason to return it in the wrapper
- the name should make clear that the close is final. (maybe `DefinitelyClose` or something like that)
- CamelCase naming convention in the new codebase. We also don't allow loose global functions like this one.

Also I am not convinced that this is deserving of its own file. One place I can think of is to merge this with `PosixIoWrappers` in `storage/write_ahead_log/log_io.h` and promote that to a file under `common`. It doesn't need to be done in this PR though.